libosmscout-map-qt/src/osmscout/MapPainterQt.cpp

```diff
@@ -480,10 +484,41 @@ namespace osmscout {
 
     QTransform tran;
 
+    // precompute character widths
+    std::vector<qreal> characterWidths(string.size());
+    for (int i=0; i<string.size(); i++) {
+      characterWidths[i]=metrics.width(string[i]);
+    }
+
     while (offset<p.length()) {
```

Comment:
Isn't this possibly too complicated? We already know the length of the way, we know the length (width) of the string. Isn't there a simpler way to check for "no angle for the next  x pixels"? This would help come around the character!=glyph problem. Also calculating the width of a character is already a medium expensive thing to do.